src/Host/Client/Impl/Transports/WebSocketMessageTransport.cs

```diff
@@ -64,8 +64,12 @@ public class WebSocketMessageTransport : WebSocketBehavior, IMessageTransport {
         public async Task SendAsync(string message, byte[] data, CancellationToken ct = default(CancellationToken)) {
             await _sendLock.WaitAsync(ct);
             try {
-                Socket.Send(message);
-                Socket.Send(data);
+                if (data.Length > 0) {
+                    Socket.Send(message);
+                    Socket.Send(data);
+                } else {
+                    throw new ArgumentException($"{nameof(data)} cannot be a 0 length array.");
```

Comment:
Why not? It would seem that a 0-sized blob would occasionally arise naturally (e.g. when uploading an empty file).